app/models/test_track/identity_session_discriminator.rb

```diff
@@ -13,6 +13,10 @@ def participate_in_online_session?
     authenticated_resource_matches_identity?
   end
 
+  def participate_in_unauthenticated_session?
+    web_context? && !controller.respond_to?(authenticated_resource_method_name, true)
```

Comment:
do these unauthenticated controllers not respond to `current_user` or does that method just return nil?